File: SDKs/Aspose.Tasks_Cloud_SDK_for_Ruby/lib/aspose_tasks_cloud/models/wbs_definition.rb
module AsposeTasksCloud
  # 
  class WBSDefinition < BaseObject
    attr_accessor :code_prefix, :generate_wbs_code, :verify_uniqueness, :code_mask_collection
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'code_prefix' => :'CodePrefix',
        
        # 
        :'generate_wbs_code' => :'GenerateWBSCode',
        
        # 
        :'verify_uniqueness' => :'VerifyUniqueness',
        
        # 
        :'code_mask_collection' => :'CodeMaskCollection'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'code_prefix' => :'String',
        :'generate_wbs_code' => :'BOOLEAN',
        :'verify_uniqueness' => :'BOOLEAN',
        :'code_mask_collection' => :'Array<WBSCodeMask>'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'CodePrefix']
        self.code_prefix = attributes[:'CodePrefix']
      end
      
      if attributes[:'GenerateWBSCode']
        self.generate_wbs_code = attributes[:'GenerateWBSCode']
      end
      
      if attributes[:'VerifyUniqueness']
        self.verify_uniqueness = attributes[:'VerifyUniqueness']
      end
      
      if attributes[:'CodeMaskCollection']
        if (value = attributes[:'CodeMaskCollection']).is_a?(Array)
          self.code_mask_collection = value
        end
      end
      
    end

  end
end

File: SDKs/Aspose.Tasks_Cloud_SDK_for_CSharp/src/Com/Aspose/Tasks/Model/WBSDefinition.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Tasks.Model {
  public class WBSDefinition {
    public string CodePrefix { get; set; }

    public bool? GenerateWBSCode { get; set; }

    public bool? VerifyUniqueness { get; set; }

    public List<WBSCodeMask> CodeMaskCollection { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class WBSDefinition {\n");
      sb.Append("  CodePrefix: ").Append(CodePrefix).Append("\n");
      sb.Append("  GenerateWBSCode: ").Append(GenerateWBSCode).Append("\n");
      sb.Append("  VerifyUniqueness: ").Append(VerifyUniqueness).Append("\n");
      sb.Append("  CodeMaskCollection: ").Append(CodeMaskCollection).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
